fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadScheduler.java

```diff
@@ -30,58 +35,96 @@
 public class RoutineLoadScheduler extends Daemon {
 
     private static final Logger LOG = LogManager.getLogger(RoutineLoadScheduler.class);
+    private static final int DEFAULT_INTERVAL_SECONDS = 10;
+
+    private RoutineLoadManager routineLoadManager;
 
-    private RoutineLoadManager routineLoadManager = Catalog.getInstance().getRoutineLoadManager();
+    @VisibleForTesting
+    public RoutineLoadScheduler() {
+        super();
+        routineLoadManager = Catalog.getInstance().getRoutineLoadManager();
+    }
+
+    public RoutineLoadScheduler(RoutineLoadManager routineLoadManager) {
+        super("Routine load", DEFAULT_INTERVAL_SECONDS * 1000);
+        this.routineLoadManager = routineLoadManager;
+    }
 
     @Override
     protected void runOneCycle() {
         try {
             process();
         } catch (Throwable e) {
-            LOG.error("failed to schedule jobs with error massage {}", e.getMessage(), e);
+            LOG.warn("Failed to process one round of RoutineLoadScheduler with error message {}", e.getMessage(), e);
         }
     }
 
-    private void process() {
+    private void process() throws UserException {
         // update
-        routineLoadManager.rescheduleRoutineLoadJob();
+        routineLoadManager.updateRoutineLoadJob();
         // get need schedule routine jobs
         List<RoutineLoadJob> routineLoadJobList = null;
         try {
             routineLoadJobList = getNeedScheduleRoutineJobs();
         } catch (LoadException e) {
-            LOG.error("failed to get need schedule routine jobs");
+            LOG.warn("failed to get need schedule routine jobs", e);
```

Comment:
no placeholder {} ? may be this line shoud replaced with Failed to get needed routine schedule jobs.

Reply:
I use the function warn(String, Throwable). It can print warn log and throwable at the same time